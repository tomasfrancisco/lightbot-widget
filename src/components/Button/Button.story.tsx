import { ThemeProviderDecorator } from "modules/ThemeProvider/ThemeProvider";
import * as React from "react";

import { storiesOf } from "@storybook/react";

import { Button } from "./Button";

storiesOf("Button", module)
  .addDecorator(ThemeProviderDecorator)
  .add("default", () => <Button touched={false}>DEFAULT</Button>)
  .add("touched", () => <Button touched={true}>touched</Button>)
  .add("negative", () => (
    <Button touched={false} negative={true}>
      DEFAULT
    </Button>
  ))
  .add("negative & touched", () => (
    <Button negative={true} touched={true}>
      touched
    </Button>
  ));
