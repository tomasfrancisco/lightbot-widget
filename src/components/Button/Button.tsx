import { WidgetThemeProps } from "lightbot-ssot/lib";
import { darken, lighten } from "polished";
import * as React from "react";
import styled, { ThemedOuterStyledProps } from "styled-components";

type ButtonProps = {
  color?: string;
  negative?: boolean;
  touched?: boolean;
};

export const Button = styled<ThemedOuterStyledProps<ButtonProps, WidgetThemeProps>, "button">(
  "button",
)`
  font-size: 14px;
  margin: 0px 1px;
  background-color: ${props => props.theme.btnPrimaryBackgroundColor};
  border: 1px solid transparent;
  border-radius: 100px;
  padding: 10px 15px;
  color: white;
  cursor: pointer;

  &:not(:only-of-type) {
    &:not(:first-of-type) {
      margin-top: 5px;
    }
  }

  &:hover {
    background-color: ${props => darken(0.1, props.color || props.theme.btnPrimaryBackgroundColor)};
    ${props => (props.negative ? "color: white" : null)};
  }

  &:active {
    background-color: ${props => darken(0.2, props.theme.btnPrimaryBackgroundColor)};
  }

  &:focus {
    outline: ${props => props.theme.outline};
  }

  ${props =>
    props.negative &&
    `
      background-color: transparent;
      border: 1px solid currentColor;
      color: ${props.color || props.theme.btnPrimaryColor};
    `};

  ${props =>
    props.touched &&
    `
      background-color: ${lighten(0.2, props.theme.btnPrimaryBackgroundColor)};
      color: white;
    `};
`;
