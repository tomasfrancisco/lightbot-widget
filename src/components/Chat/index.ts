export * from "./Chat";
export * from "./ChatHeader";
export * from "./ChatBody";
export * from "./ChatFooter";
