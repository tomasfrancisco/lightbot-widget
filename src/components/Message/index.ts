export * from "./Message";
export * from "./MessageGroup";
