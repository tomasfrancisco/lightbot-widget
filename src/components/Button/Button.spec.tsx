import { theme } from "@lightbot/widget-configuration";
import { Button } from "components/Button";
import { shallow } from "enzyme";
import * as React from "react";

describe("components/Button", () => {
  let component;

  beforeEach(() => {
    component = shallow(<Button theme={theme} touched={false} />);
  });

  it("renders", () => {
    expect(component).toMatchSnapshot();
  });
});
