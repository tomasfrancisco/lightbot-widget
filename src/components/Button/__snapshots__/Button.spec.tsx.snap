// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Button renders 1`] = `
<button
  className="sc-bdVaJa ffDxtS"
/>
`;
